lib/qmp.ml

```diff
@@ -143,11 +148,22 @@ let message_of_string str =
   in
 
   let event json =
+    let event_data event data =
+      let rtc_offset data = data |> U.member "offset" |> U.to_int in
+      match event with
+        | "RTC_CHANGE"      -> data |> rtc_offset |> fun x -> Some (RTC_CHANGE x)
+        | "POWERDOWN"       -> None
+        | "BLOCK_IO_ERROR"  -> None
+        | x -> Printf.fprintf stderr "unsupported event [%s]\n" x; None
```

Comment:
Hi @mg12 , thanks for your comments. There are some others `Events` send from QEMU, like `VNC_CONNECTED`, `VNC_INITIALIZED`, `SHUTDOWN`, `DEVICE_DELETED`. I think they are normal events, no need to `failwith`.
And looks like `failwith` just return an error json, not an exception, here is the log:
```
Sep 12 04:09:27 xrtuk-11-38 xenopsd-xc: [debug|xrtuk-11-38|2 ||xenops] Got non-event message, domain-13: {"error":{"class":"JSONParsing","desc":"(Failure \"unsupported event [VNC_CONNECTED]\\n\") in {\"timestamp\": {\"seconds\": 1505189367, \"microseconds\": 249993}, \"event\": \"VNC_CONNECTED\", \"data\": {\"server\": {\"auth\": \"none\", \"family\": \"ipv4\", \"service\": \"5901\", \"host\": \"127.0.0.1\", \"websocket\": false}, \"client\": {\"family\": \"ipv4\", \"service\": \"34066\", \"host\": \"127.0.0.1\", \"websocket\": false}}}\r","data":{}}}
```

Reply:
If we want to ignore event data by default and only handle for some events, then I think we should only list RTC_CHANGE and ignore the rest:
``` 
+        | "RTC_CHANGE"      -> data |> rtc_offset |> fun x -> Some (RTC_CHANGE x)
+        (* ignore data for other events *)
+        | _ -> None
```
This may be justifiable because we are only interested in the RTC_CHANGE event, and we are not going to handle the other events anyway. We can add other useful event data in the future. @mseri what do you think?

I think you should still change device.ml to capture a generic e as above, because event_data() in the future can fail for unpredictable reasons and we do not want to miss qmp events in xenopsd.

Reply:
I agree with @mg12 comment. We should ignore the rest and add the necessary data handlers only when needed, but we should still capture all exceptions

Reply:
Thanks, updated.
For `qmp eexception`, submitted commit : https://github.com/xapi-project/xenopsd/pull/384/commits/b3c6b731a577111e9e6591e78d5f43c182d13ea0